runtime/src/main/java/jp/co/soramitsu/runtime/network/RuntimeCall.kt

```diff
@@ -0,0 +1,35 @@
+package jp.co.soramitsu.runtime.network
+
+import io.emeraldpay.polkaj.scale.ScaleCodecReader
+import java.math.BigInteger
+import jp.co.soramitsu.fearless_utils.extensions.toHexString
+import jp.co.soramitsu.fearless_utils.runtime.AccountId
+import jp.co.soramitsu.fearless_utils.scale.dataType.uint128
+import jp.co.soramitsu.fearless_utils.wsrpc.request.runtime.RuntimeRequest
+
+sealed class RuntimeCall<T> {
+
+    abstract val path: String
+    abstract val args: ByteArray
+    abstract fun parseResult(resultBytes: ByteArray): T
+
+    sealed class NominationPoolsApi<T> : RuntimeCall<T>() {
+
+        class PendingRewards(accountId: AccountId) : NominationPoolsApi<BigInteger>() {
+            override val path: String = "NominationPoolsApi_pending_rewards"
+            override val args: ByteArray = accountId
+
+            override fun parseResult(resultBytes: ByteArray): BigInteger {
+                val reader = ScaleCodecReader(resultBytes)
+                return uint128.read(reader)
+            }
+        }
+    }
+}
+
+class RuntimeCallRequest(runtimeCallMethod: String, args: String) : RuntimeRequest(
+    method = "state_call",
+    params = listOf(runtimeCallMethod, args)
+)
+
+fun RuntimeCall<*>.toRequest(): RuntimeCallRequest = RuntimeCallRequest(path, args.toHexString(true))
```

Comment:
`named params` for boolean parameter?

Reply:
![image](https://user-images.githubusercontent.com/19205494/200837047-fef49c9c-0536-45a6-b203-a0756c218827.png)
This is not enough? :)

Reply:
It's sometimes lagging, and it's difficult to check code on PRs. I prefer to use named params if there is no named variable always. But it's up to you :)